sqlmesh/integrations/dlt.py

```diff
@@ -73,15 +78,32 @@ def generate_dlt_models_and_settings(
         grain = f"\n  grain ({', '.join(primary_key)})," if primary_key else ""
         incremental_model_name = f"{dataset}_sqlmesh.incremental_{table_name}"
 
-        incremental_model_sql = generate_incremental_model(
-            incremental_model_name,
-            select_columns,
-            grain,
-            dataset + "." + table_name,
-            dialect,
-        )
-
-        sqlmesh_models.add((incremental_model_name, incremental_model_sql))
+        if load_id:
+            load_key = "c." + load_id
+            parent_table = None
+
+            # Handling for nested tables: https://dlthub.com/docs/general-usage/destination-tables#nested-tables
+            if load_id == "_dlt_parent_id" and "_dlt_parent_id" in dlt_columns:
+                if parent_table := table["parent"]:
+                    if (
+                        parent_table in dlt_tables
+                        and "_dlt_id" in dlt_tables[parent_table]["columns"]
+                    ):
+                        load_key = "p._dlt_load_id"
+                        parent_table = dataset + "." + parent_table
+                    else:
+                        break
```

Comment:
[Nit] consider flipping the branches so that you can dedent L93 and reduce nesting a bit.